Adding nightly IPA tests

SilleBille opened the pull request:
2 containers are being spawned (1 of them will be removed in future)
- 1st container downloads the PKI rpms from COPR
- ipa-docker-test-runner spawns a new container and installs \
	the latest PKI rpms -- the tool will be removed in future

Signed-off-by: Dinesh Prasanth M K <dmoluguw@redhat.com>

SilleBille committed:
Adding nightly IPA tests

- 1st container downloads the PKI rpms from COPR
- ipa-docker-test-runner spawns a new container and installs \
	the latest PKI rpms -- the tool will be removed in future

Signed-off-by: Dinesh Prasanth M K <dmoluguw@redhat.com>

SilleBille commented:
Travis is removing the encrypted variable since this is a PR. I guess the patch should work fine -- it can only be tested after merging! (unless, you have any other tweaks)

edewata reviewed (approved):
The new code looks good, but there's a failure from the existing code.

SilleBille committed:
Increasing wait time and moving script to file

- `travis_wait` has been added since `copr build`
might become stale if COPR build takes longer
time to build

- The build script is moved into file since `travis_wait`
removes the quotes and the command behaves abnormally

Signed-off-by: Dinesh Prasanth M K <dmoluguw@redhat.com>

SilleBille commented:
@edewata I setup the encrypted variable onto my personal repo.  The script has passed the test. The results can be viewed here: 

The reason for failure in this PR is because of the encrypted variable (inside travis repo) isn't exported. 
https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions
"Pull requests sent from forked repositories do not have access to encrypted variables or data."

SilleBille commented:
@edewata I setup the encrypted variable onto my personal repo. The script has passed the test. The results can be viewed here: https://travis-ci.org/SilleBille/pki-nightly-test/jobs/411957011

The reason for failure of the existing code is because of the encrypted variable (inside travis repo) isn't exported.
https://docs.travis-ci.com/user/pull-requests/#pull-requests-and-security-restrictions
"Pull requests sent from forked repositories do not have access to encrypted variables or data."

Also, the other issue is that sometimes COPR build takes >10 mins. This leads to termination of the `copr build` command by travis. When I try to set up `travis_wait`, docker behaves unexpectedly. I'll fix this in the next patch which I'll be submitting on Monday.

SilleBille commented:
The CI results for PR fails due to uninjected environment variables.

The other issue is fixed in the last commit by adding `travis_wait` and moving all COPR related scripts to sh file. Build passes successfully in personal Travis build: https://travis-ci.org/SilleBille/pki-nightly-test/builds/412741227
